### PR TITLE
Upgrade Metronome base to Marathon 1.3.13

### PR DESCRIPTION
**Pull request by kensipe**
The base Marathon version of Metronome is RC-1.   Looking to upgrade to a released version of Marathon at 1.3.13.   
Almost all of the changes are based with the change between RC5 and RC6:
https://github.com/mesosphere/marathon/compare/v1.2.0-RC5...v1.2.0-RC6

I'm not confident in a couple areas which I will call out in a review for other reviewers to be aware.

All the unit and integration tests run successfully.

### CODE REVIEW

File: jobs/src/main/scala/dcos/metronome/scheduler/SchedulerConfig.scala [hunk 1]
```diff
@@ -2,11 +2,13 @@ package dcos.metronome.scheduler
 
 import dcos.metronome.repository.impl.kv.ZkConfig
 import mesosphere.marathon.AllConf
+import mesosphere.marathon.core.task.termination.TaskKillConfig
 
 import scala.concurrent.duration.FiniteDuration
 
 trait SchedulerConfig extends ZkConfig {
   def scallopConf: AllConf
+  def taskKillConfig: TaskKillConfig
```
Comment (kensipe) [inline jobs/src/main/scala/dcos/metronome/scheduler/SchedulerConfig.scala]: We need the KillConfig but I'm not familiar with Scallop and the approach of having multiple config traits.   The metronome way was to have traits that are not extensions of scallop...  looking for advice here.

File: jobs/src/main/scala/dcos/metronome/scheduler/SchedulerModule.scala [hunk 2]
```diff
@@ -50,8 +50,7 @@ class SchedulerModule(
   private[this] lazy val shutdownHooks = ShutdownHooks()
   private[this] lazy val actorsModule = new ActorsModule(shutdownHooks, actorSystem)
 
-  private[this] lazy val eventModule: EventModule = new EventModule(scallopConf)
-  private[this] lazy val eventBus: EventStream = eventModule.provideEventBus(actorSystem)
+  private[this] lazy val eventBus: EventStream = actorSystem.eventStream
```
Comment (kensipe) [inline jobs/src/main/scala/dcos/metronome/scheduler/SchedulerModule.scala]: we removed eventModule completely... marathon made it hard to use from marathon.. however the underlying impl was to get the eventStream.

File: jobs/src/main/scala/dcos/metronome/scheduler/SchedulerModule.scala [hunk 3]
```diff
@@ -136,8 +144,7 @@ class SchedulerModule(
       frameworkIdUtil,
       leaderInfo,
       actorSystem,
-      scallopConf,
-      schedulerCallbacks
```
Comment (kensipe) [inline jobs/src/main/scala/dcos/metronome/scheduler/SchedulerModule.scala]: scheduler call back was completely removed.

**Comment by kensipe** [comment]
The velocity failure isn't real... I'm not sure what that is but we need to fix the metronome env.

File: jobs/src/main/scala/dcos/metronome/utils/glue/MarathonImplicits.scala [hunk 4]
```diff
@@ -68,11 +68,8 @@ object MarathonImplicits {
   implicit class JobSpecToContainer(val jobSpec: JobSpec) extends AnyVal {
     def toContainer: Option[Container] = {
       jobSpec.run.docker match {
-        case Some(dockerSpec) => Some(Container(
-          `type` = mesos.Protos.ContainerInfo.Type.DOCKER,
-          docker = Some(Container.Docker(
-            image = dockerSpec.image
-          )),
+        case Some(dockerSpec) => Some(Container.Docker(
+          image = dockerSpec.image,
```
Comment (timcharper) [inline jobs/src/main/scala/dcos/metronome/utils/glue/MarathonImplicits.scala]: Does 1.3.x not have support for UCR?

File: jobs/src/test/scala/dcos/metronome/jobspec/impl/JobSpecSchedulerActorTest.scala [hunk 5]
```diff
@@ -66,6 +66,22 @@ class JobSpecSchedulerActorTest extends TestKit(ActorSystem("test")) with FunSui
...
+
+    When("The actor is created")
+    actor ! StartJob
+
+    Then("The next run is rescheduled")
+    eventually(actor.underlyingActor.scheduledAt should be(Some(nextRun)))
+    actor.underlyingActor.nextSchedule should be (defined)
+    system.stop(actor)
+  }
+
```
Comment (timcharper) [inline jobs/src/test/scala/dcos/metronome/jobspec/impl/JobSpecSchedulerActorTest.scala]: 👏 👏 👏

File: project/Build.scala [hunk 6]
```diff
@@ -131,7 +131,7 @@ object Build extends sbt.Build {
       val AsyncAwait = "0.9.6-RC2"
       val ScalaTest = "2.1.7"
       val MacWire = "2.2.2"
-      val Marathon = "1.2.0-RC1"
+      val Marathon = "1.3.0"
```
Comment (timcharper) [inline project/Build.scala]: Why not 1.3.13?

File: jobs/src/main/scala/dcos/metronome/scheduler/SchedulerConfig.scala (hunk 1 above)
Comment (timcharper) [reply jobs/src/main/scala/dcos/metronome/scheduler/SchedulerConfig.scala]: taskKillConfig I think is derived from multiple scallop options. Does it compile? Does it work?

File: jobs/src/main/scala/dcos/metronome/scheduler/SchedulerModule.scala (hunk 2 above)
Comment (timcharper) [reply jobs/src/main/scala/dcos/metronome/scheduler/SchedulerModule.scala]: Seems better

File: jobs/src/main/scala/dcos/metronome/scheduler/impl/NotifyOfTaskStateOperationStep.scala [hunk 7]
```diff
@@ -34,7 +33,8 @@ class NotifyOfTaskStateOperationStep(eventBus: EventStream, clock: Clock) extend
     (taskChanged.stateOp, taskChanged.stateChange) match {
 
       // A Mesos status update disregarding the effect
-      case (TaskStateOp.MesosUpdate(task, WithMesosStatus(mesosStatus), _), _) => Some(TaskState(mesosStatus))
+      case (TaskStateOp.MesosUpdate(task, marathonTaskStatus, mesosStatus, _), _) => Some(TaskState(mesosStatus))
```
Comment (timcharper) [inline jobs/src/main/scala/dcos/metronome/scheduler/impl/NotifyOfTaskStateOperationStep.scala]: Seems fine :) Get rid of the comment below?

File: version.sbt [hunk 8]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.2.4"
+version in ThisBuild := "0.2.5"
```
[the remark this answers is not in the document]
Comment (alenkacz) [reply version.sbt]: +1

File: jobs/src/main/scala/dcos/metronome/utils/glue/MarathonImplicits.scala (hunk 4 above)
Comment (meichstedt) [reply jobs/src/main/scala/dcos/metronome/utils/glue/MarathonImplicits.scala]: It does. So far, Metronome only supports Docker via its API, but adding UCR as an option sould be a good next step after bumping.

File: project/Build.scala (hunk 6 above)
Comment (meichstedt) [reply project/Build.scala]: It should eventually be 1.3.13. I think it makes sense to first have everything adjusted to 1.3.0 and compiling, and then bump further.

As a third step I'd be inclined to bump Play to 2.5.18, and eventually to 2.6

File: version.sbt (hunk 8 above)
[the remark this answers is not in the document]
Comment (meichstedt) [reply version.sbt]: +1

**Comment by kensipe** [comment]
@timcharper we do have a suite of SI tests... I haven't run them yet... the unit and integration tests are green

**Comment by kensipe** [comment]
@timcharper @alenkacz @meichstedt This work is complete now.   It passes unit and integration tests.   I did some light testing locally as well.   It has:

- Integration with marathon 1.13.0
- integration of config to play for taskkill 
- update to latest stable of AsyncAwait, ScalaTest, Play and WixAccord
- updated version to minor release of 0.3

notes:  Issues with upgrading to a later version of ScalaTest and Play

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

File: jobs/src/main/scala/dcos/metronome/scheduler/impl/NotifyOfTaskStateOperationStep.scala (hunk 7 above)
Comment (alenkacz) [reply jobs/src/main/scala/dcos/metronome/scheduler/impl/NotifyOfTaskStateOperationStep.scala]: marathonTaskStatus is not used, so "_" would be enough...

Comment (alenkacz) [reply jobs/src/main/scala/dcos/metronome/scheduler/impl/NotifyOfTaskStateOperationStep.scala]: maybe in the future we can use MarathonTaskStatus and get rid of Metronome mapping... ?

I think that what is in TaskState in metronome is probably very similar to what is happening in MarathonTaskStatus?

**Comment by kensipe** [comment]
All PR comments addressed.  Planning on merging after tests.

**Pull request merged**